### PR TITLE
PR for review regarding changes to upstream

### PR DESCRIPTION
**Pull request by Licenser**
This PR is to review changes the Ky/community/hex version of cuttlefish has and prepare for a 'clean' and approved merge to upstream cuttlefish.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
 
-{erl_opts, [warnings_as_errors, {parse_transform, lager_transform}, debug_info, warn_untyped_record]}.
+{erl_opts, [warnings_as_errors,
+            {parse_transform, lager_transform},
+            debug_info,
+            warn_untyped_record]}.
 
-{eunit_opts, [verbose]}.
-{cover_enabled, true}.
+{deps, [getopt, lager]}.
```
Comment (Licenser) [inline rebar.config]: Versioned dependencies would be nice here to make it clear which versions we require and prevent accidental upgrades.

File: .gitignore [hunk 2]
```diff
@@ -1,15 +1,15 @@
+.rebar3
+_build
 .DS_Store
 generated.config
 generated.conf
 ebin/*
 .eunit
-src/conf_parse.erl
```
Comment (Licenser) [inline .gitignore:6]: Why do we commit a generated file to the repo?

File: src/cuttlefish.app.src [hunk 3]
```diff
@@ -7,5 +7,9 @@
                   kernel,
                   stdlib
                  ]},
-  {env, []}
+  {env, []},
+
+  {maintainers,[]},
+  {licenses,["Apache"]},
+  {links,[{"Github","https://github.com/tsloughter/cuttlefish"}]}
```
Comment (Licenser) [inline src/cuttlefish.app.src:14]: have this point here or to the basho repo so people find the right repo when clicking on the link in hex?

File: src/cuttlefish_duration_parse.erl [hunk 4]
```diff
@@ -0,0 +1,338 @@
+-module(cuttlefish_duration_parse).
```
Comment (Licenser) [inline src/cuttlefish_duration_parse.erl:1]: why commit a generated file?

File: src/cuttlefish_unit.erl [hunk 5]
```diff
@@ -1,7 +1,7 @@
 -module(cuttlefish_unit).
 
 -include_lib("eunit/include/eunit.hrl").
--compile(export_all).
+-compile([nowarn_export_all, export_all]).
```
Comment (Licenser) [inline src/cuttlefish_unit.erl:4]: shouldn't this live under `test` if it is a unit test file?

**Comment by ghost** [comment]
Any indication of timescale when this will get merged upstream? Or is it the case this will become the canonical repository?

**Comment by Licenser** [comment]
I'm not sure. I've had zero luck so far with any upstream merges even for simpler libraries with fewer changes. (perhaps no one saw them just like I missed this PR :P) My personal preference would be to have one canonical place for most if not all things just so people don't get confused.

`riak_core_ng` might be a special case as I'm unsure if it is good to merge upstream of suposed to be merged upstream or wanted to be merged upstream, but the whole set of libs would be nice to reduce the number of forks.

**Pull request closed without merging**